minard/channeldb.py

```diff
@@ -303,6 +303,62 @@ def get_discriminator_threshold(crate, slot, channel):
 
     return threshold
 
+def get_all_thresholds(run):
+
+    conn = engine.connect()
+
+    zero = [[0 for i in range(16)] for j in range(19)]
+    thr = [[0 for i in range(16)] for j in range(19)]
+    disc = [9999]*9728
+
+    # Select most recent zdisc with ecalid field
+    result = conn.execute("select crate, slot, zero_disc from zdisc where "
+                           "(ecalid <> '') is True "
+                           "order by timestamp DESC limit 304")
+   
+    if result is None:
+        return None
+
+    rows = result.fetchall()
+    for crate, slot, zero_disc in rows:
+        zero[crate][slot] = zero_disc
+
+    result = conn.execute("select crate, slot, vthr from detector_state where "
+                          "run = %s order by crate, slot" % run)
```

Comment:
Should pass run as an argument instead of using python string formatting to prevent sql injection attacks.